force_bdss/data_sources/base_data_source_factory.py

```diff
@@ -70,7 +70,21 @@ def create_data_source(self):
         BaseDataSource
             The specific instance of the generated DataSource
         """
-        return self.data_source_class(self)
+
+        try:
+            data_source = self.data_source_class(self)
```

Comment:
These try statements were being used in multiple places upon calling `factory.data_source_class`, so it seemed sensible to include them inside the method.